File: Solutions/JunkyDragon/Game Shop/Part2/Weapon.java
package com.github.junkydragon;

/**
 * Created by Administrator on 2017-05-23.
 */
public class Weapon extends Item{
    private int damage;
    public Weapon(){super();}
    public Weapon(String name, String description, float weight, int value, int damage) {
        super(name, description, weight, value);
        this.damage = damage;
    }
    public void Describe() {
        super.Describe();
        System.out.println("Damage\t\t\t = " +this.damage);
    }
}

File: Solutions/JunkyDragon/Game Shop/Part2/Weapon.kt
package org.junkydragon.item

/**
 * Created by Administrator on 2017-05-24.
 */

class Weapon(name: String, description: String, weight: Double, value: Int, val damage:Int):Item(name, description, weight, value){
    override fun About() {
        println("Name\t\t\t= $name")
        println("Description\t\t= $description")
        println("Weight\t\t\t= $weight")
        println("Value\t\t\t= $value")
        println("Damage\t\t\t= $damage")
    }
}